source/common/grpc/common.cc

```diff
@@ -206,6 +206,19 @@ uint64_t Common::grpcToHttpStatus(Status::GrpcStatus grpc_status) {
   }
 }
 
+void Common::sendLocalReply(
+    std::function<void(Http::HeaderMapPtr&& headers, bool end_stream)> encode_headers,
+    Status::GrpcStatus grpc_status, const std::string& message) {
+  Http::HeaderMapPtr response_headers{new Http::HeaderMapImpl{
+      {Http::Headers::get().Status, std::to_string(enumToInt(Http::Code::OK))},
+      {Http::Headers::get().ContentType, Http::Headers::get().ContentTypeValues.Grpc},
+      {Http::Headers::get().GrpcStatus, std::to_string(enumToInt(grpc_status))}}};
+  if (!message.empty()) {
+    response_headers->insertGrpcMessage().value(message);
```

Comment:
We could just punt on this for now and leave it as a TODO if not needed. FWIW, I don't recall percent encoding being mandatory, but maybe this has changed, since the wire format at https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md specifies it, with some confusing language (e.g. UTF-8 _followed_ by percent encoding?!). @lizan for thoughts here.

Reply:
Yeah I would just TODO it. I don't think it's explicitly required at the h2 level AFAIK. Maybe it is for gRPC.

Reply:
Yeah I think a TODO here is fine (open an issue to track it) for now. We don't have local messages requires % encoding, also it should just work since h2 doesn't require this.

Reply:
Added TODO, which is now in http/utility.cc